### PR TITLE
feat: implement server side logging

### LINKED ISSUE #6: implement server side logging

### PR DESCRIPTION
**Pull request by luisbedoia**

### CODE REVIEW

**Review by neeleshpoli — APPROVED** [review]
Changes look good 👍

**Review by TBS1996 — APPROVED** [review]
neat, great job!

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "rust-analyzer.linkedProjects": ["./Cargo.toml"],
+  "editor.inlayHints.enabled": "offUnlessPressed",
```
Comment (henriquekirchheck) [inline .vscode/settings.json]: I understand the cargo features, but why are these two lines needed? I don't think they should solve any specific problems, do they?

File: src/server.rs [hunk 2]
```diff
@@ -172,17 +174,23 @@ async fn pair_handler(
 }
 
 pub async fn run() {
+    env::set_var("RUST_BACKTRACE", "1");
+    env::set_var("RUST_LOG", "tower_http=debug,ocean_chat=debug");
+    tracing_subscriber::fmt::init();
```
Comment (henriquekirchheck) [inline src/server.rs]: I'd prefer to use something like this to configure the layer, it should make it possible to overwrite it with environment variables if needed 

```rust
tracing_subscriber::registry()
        .with(
            tracing_subscriber::EnvFilter::try_from_default_env().unwrap_or_else(|_| {
                "ocean_chat=debug,tower_http=debug,axum::rejection=trace".into()
            }),
        )
        .with(tracing_subscriber::fmt::layer())
        .init();
```

**Review by henriquekirchheck — COMMENTED** [review]
There should probably be more information in each log, like user id or in the case of the starting server log the server address would be useful

File: .vscode/settings.json (hunk 1 above)
Comment (luisbedoia) [reply .vscode/settings.json]: The line 2: 
"Disable project auto-discovery in favor of explicitly specified set of projects.
Elements must be paths pointing to Cargo.toml, rust-project.json, .rs files (which will be treated as standalone files) or JSON objects in rust-project.json format."

Whit line 3:
image

without line 3: 
IMO makes the code more verbose and less readable.
image

File: src/server.rs (hunk 2 above)
Comment (luisbedoia) [reply src/server.rs]: nice catch.

File: .vscode/settings.json (hunk 1 above)
Comment (henriquekirchheck) [reply .vscode/settings.json]: I quite like the hints, so I think this should be placed in the global vscode config so everyone can choose based on their own preferences, not enforce a purely visual change to everyone in the project.

Comment (luisbedoia) [reply .vscode/settings.json]: ok.

**Review by henriquekirchheck — APPROVED** [review]
LGTM

**Pull request merged**